### PR TITLE
[Packager] Generate module name seperated by '/' on Windows.

### PR DESCRIPTION
**Pull request by tdzl2003**
I think packager on different platform should generate same output if possible. So packager should replace '\\' in module name with '/' on Windows.

### CODE REVIEW

**Comment by mkonicek** [comment]
@martinbigio What do you think?

**Comment by martinbigio** [comment]
should you also update the the on the constructor? `this.path = path.resolve(file);`

Take a look also on  https://github.com/facebook/react-native/pull/2829. I don't have a windows computer to make sure either this or that is sufficient to fix the path issue

**Comment by tdzl2003** [comment]
#2829 still causing errors on my machine. It changed module define, but not changing path in require  statement。

I don't think change "path" property is necessary. That may effect to many places. File path on windows can still performed with '\\', only the module name used in __d and require should replace '\\' to '/'

**Comment by martinbigio** [comment]
The change looks fine to me. I wonder if the mentioned PR (which we've already landed) wouldn't be necessary with this. Could you if we could keep the internals as they used to be (reverting that diff)?

**Comment by tdzl2003** [comment]
No it's not necessary. 

#2829 still causing errors on my machine. It changed module define and dependencies, but not changing path in require statement。

**Comment by martinbigio** [comment]
do you mind including the revert of that on this PR?

**Comment by tdzl2003** [comment]
#2829 merged and reverted.

File: packager/react-packager/src/DependencyResolver/Module.js [hunk 1]
```diff
@@ -56,7 +56,7 @@ class Module {
               return this.path;
             }
 
-            return path.join(name, path.relative(p.root, this.path));
+            return path.join(name, path.relative(p.root, this.path)).replace(/\\/g, '/');
```
Comment (mkonicek) [inline packager/react-packager/src/DependencyResolver/Module.js]: Is there any way `path.join` can return `null` or `undefined`?

**Comment by mkonicek** [comment]
Cool! Is this good to go now?

File: packager/react-packager/src/DependencyResolver/Module.js (hunk 1 above)
Comment (martinbigio) [inline packager/react-packager/src/DependencyResolver/Module.js]: not if the parameters are defined

**Comment by martinbigio** [comment]
@facebook-github-bot shipit

**Pull request closed without merging**